src/controller/ReadInteraction.h

```diff
@@ -151,13 +161,15 @@ CHIP_ERROR SubscribeAttribute(
         nullptr,
     typename TypedReadAttributeCallback<DecodableAttributeType>::OnResubscriptionAttemptCallbackType onResubscriptionAttemptCb =
         nullptr,
-    bool fabricFiltered = true, bool keepPreviousSubscriptions = false, const Optional<DataVersion> & aDataVersion = NullOptional)
+    typename detail::SubscriptionOnDoneCallback onDoneCb = nullptr, bool fabricFiltered = true,
+    bool keepPreviousSubscriptions = false, const Optional<DataVersion> & aDataVersion = NullOptional)
```

Comment:
From an API standpoint post 1.0, do we want to move new optional arguments to the end? Or make an API breaking change like this? I know API changes are acceptable and are not held to the same standard as over-the-wire changes, but wanted to better understand the guidance here.

@andreilitvin thoughts?

Reply:
That's a good question. I'd also like to know if there's guidance or general consensus on this. I can certainly move it to the back if that's desired.

Reply:
I have moved the new argument to the end, so there should be issues with breaking API.

Reply:
I think we must accept some API changes will happen in master.

Having said that, this converts a 13-argument method to a 14 one. This is not maintainable and makes api incompatibility even worse (even though the CB has a default, so we don't really break clients)

I would vote that if we touch this, we should make the arguments take some builder-struct. I don't think it would cost much more stack or code, but would increase readability a lot.